### PR TITLE
Move FieldDescription related classes to its folder

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

In order to provide a smooth upgrade path to the users, this PR tries to deprecate `FieldDescription` related classes, I've followed https://dev.to/greg0ire/how-to-deprecate-a-type-in-php-48cf, so please @greg0ire if you can take a look would be great.

I am targeting this branch, because these changes are hopefully BC.

## Changelog

```markdown
### Added
- Added `Sonata\AdminBundle\FieldDescription\BaseFieldDescription` class.
- Added `Sonata\AdminBundle\FieldDescription\FieldDescriptionCollection` class.
- Added `Sonata\AdminBundle\FieldDescription\FieldDescriptionInterface` interface.
- Added `Sonata\AdminBundle\FieldDescription\FieldDescriptionRegistryInterface` interface.
### Deprecated
- Deprecated `Sonata\AdminBundle\Admin\BaseFieldDescription` class.
- Deprecated `Sonata\AdminBundle\Admin\FieldDescriptionCollection` class.
- Deprecated `Sonata\AdminBundle\Admin\FieldDescriptionInterface` interface.
- Deprecated `Sonata\AdminBundle\Admin\FieldDescriptionRegistryInterface` interface.
```

### CODE REVIEW

**Comment by wbloszyk** [comment]
Simmilar to work done in CoreBundle.

**Review by wbloszyk — APPROVED** [review]
Simmilar to work done in CoreBundle.

File: src/Builder/BuilderInterface.php [hunk 1]
```diff
@@ -29,3 +29,5 @@ interface BuilderInterface
      */
     public function fixFieldDescription(AdminInterface $admin, FieldDescriptionInterface $fieldDescription);
 }
+
+class_exists(FieldDescriptionInterface::class);
```
Comment (greg0ire) [inline src/Builder/BuilderInterface.php:33]: This will work, but using `interface_exists` would feel less weird.

**Review by greg0ire — COMMENTED** [review]
It looks correct 👍

**Review by VincentLanglet — APPROVED** [review]
Can you do a 3.x merge into master after this merge ? @franmomu

**Comment by franmomu** [comment]
Sure thing.

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thank you @franmomu

**Pull request merged**